environments/openshift/jaeger.jsonnet

```diff
@@ -0,0 +1,93 @@
+local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
+
+local app =
+  (import '../kubernetes/jaeger.libsonnet') + {
+    jaeger+:: {
+      namespace:: '${NAMESPACE}',
+      image:: '${IMAGE}:${IMAGE_TAG}',
+      replicas:: '${{REPLICAS}}',
+
+      queryService+: {
+        metadata+: {
+          annotations+: {
+            'service.alpha.openshift.io/serving-cert-secret-name': 'querier-tls',
+          },
+        },
+        spec+: {
+          ports+: [
+            { name: 'https', port: 16687, targetPort: 16687 },
+          ],
+        },
+      },
+
+      local deployment = k.apps.v1.deployment,
+      local volume = deployment.mixin.spec.template.spec.volumesType,
+      local container = deployment.mixin.spec.template.spec.containersType,
+      local volumeMount = container.volumeMountsType,
+
+      deployment+: {
+        spec+: {
+          template+: {
+            spec+: {
+              containers+: [
+                container.new('proxy', '${PROXY_IMAGE}:${PROXY_IMAGE_TAG}') +
+                container.withArgs([
+                  '-provider=openshift',
+                  '-https-address=:%d' % $.jaeger.queryService.spec.ports[1].port,
+                  '-http-address=',
+                  '-email-domain=*',
+                  '-upstream=http://localhost:%d' % $.jaeger.queryService.spec.ports[0].port,
+                  '-openshift-service-account=prometheus-telemeter',
+                  '-openshift-sar={"resource": "namespaces", "verb": "get", "name": "${NAMESPACE}", "namespace": "${NAMESPACE}"}',
+                  '-openshift-delegate-urls={"/": {"resource": "namespaces", "verb": "get", "name": "${NAMESPACE}", "namespace": "${NAMESPACE}"}}',
+                  '-tls-cert=/etc/tls/private/tls.crt',
+                  '-tls-key=/etc/tls/private/tls.key',
+                  '-client-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token',
+                  '-cookie-secret-file=/etc/proxy/secrets/session_secret',
+                  '-openshift-ca=/etc/pki/tls/cert.pem',
+                  '-openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt',
+                  '-skip-auth-regex=^/metrics',
```

Comment:
I copied that proxy sidecar from other components, so this might not be totally on point for Jaeger. But I can easily remove that line for now.
Adding the jaeger-mixin will be a separate PR.